File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        #bfs to invert levels , use collection.deque
        


        # iterative dfs, preorder traversal
        
        stack = []
        
        if root:
            stack.append(root)
        
        while len(stack) > 0:
            node = stack.pop()
            
            if node.left or node.right:
                node.left, node.right = node.right, node.left
                
                if node.left:
                    stack.append(node.left)
                if node.right:
                    stack.append(node.right)
                    
        return root
            
        
            
        

File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        
        Queue<TreeNode> queue = new LinkedList<>();
        
        if (root!= null){
            queue.add(root);
        }
        
        while(!queue.isEmpty()){
            
            TreeNode node = queue.remove();
            TreeNode left = node.left;
            TreeNode right = node.right;
            
            TreeNode tmp = left;
            node.left = right;
            node.right = tmp;
            
            if (node.left != null){
                queue.add(node.left);
            }
            
            if(node.right != null){
                queue.add(node.right);
            }
        }
        
        
        return root;

        
    }
}